calicoctl/commands/resources.go

```diff
@@ -41,15 +38,15 @@ const (
 	actionCreate
 	actionUpdate
 	actionDelete
-	actionList
+	actionGetOrList
 )
 
 // Convert loaded resources to a slice of resources for easier processing.
 // The loaded resources may be a slice containing resources and resource lists, or
 // may be a single resource or a single resource list.  This function handles the
 // different possible options to convert to a single slice of resources.
-func convertToSliceOfResources(loaded interface{}) []unversioned.Resource {
-	r := []unversioned.Resource{}
+func convertToSliceOfResources(loaded interface{}) []runtime.Object {
+	res := []runtime.Object{}
 	log.Infof("Converting resource to slice: %v", loaded)
 
 	switch reflect.TypeOf(loaded).Kind() {
```

Comment:
I actually think this processing can be simplified a lot now....

We know that the interface passed in will either be:
-  A runtime.Object
-  A slice of runtime.Objects

And then for each runtime.Object, we know that it will either be
-  A ResourceObject
-  A ResourceList

That's probably much "safer" processing than all the Ptr/Struct stuff which we originally needed because our v1 resources and lists didn't implement an interface.

Have you put this code in it's current guide through a thorough test, testing applying a List or a single resource, or a slice of Lists or slice of Lists and resources etc.  ?

Reply:
(maybe simplified is not the right word - but certainly made easier to understand).